File: Treino/maiuscula.c
#include <stdio.h>

int is_fim(char frase[]);

int verifica(char frase[]);

int main(){
    char frase[255];
    fgets(frase, 255, stdin);
    while(!is_fim(frase)){
        printf("%d\n", verifica(frase));
        fgets(frase, 255, stdin);   
    }
    return 0;
}
int is_fim(char frase[]){
    int tamanho = 0;
    for(int i = 0;frase[i] != '\0' && frase[i] != '\n';i++){
        tamanho++;
    }
    return tamanho == 3 && frase[0] == 'F' && frase[1] == 'I' && frase[2] == 'M';
}
int verifica(char frase[]){
   int maiuscula = 0;
   for(int i = 0;frase[i] != '\0' && frase[i] != '\n';i++){
        if(frase[i] >= 'A' && frase[i] <= 'Z'){
            maiuscula++;
        }
   }
   return maiuscula;

}

File: Treino/Maiuscula.java
import java.util.Scanner;

class Maiuscula{
    public static boolean isFim(String frase){
        return frase.length() == 3 && frase.charAt(0) == 'F' && frase.charAt(1) == 'I' && frase.charAt(2) == 'M';
    }
    public static int verifica(String frase){
        int maiuscula = 0;

        for(int i = 0;i < frase.length();i++){
            if(frase.charAt(i) >= 'A' && frase.charAt(i) <= 'Z'){
                maiuscula++;
            }
        }
        return maiuscula;
    }
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        String frase = sc.nextLine();

        while(!isFim(frase)){
            System.out.println(verifica(frase));   
            frase = sc.nextLine();
        }
        sc.close();
    }
}
